php-checks/src/test/resources/checks/UnusedPrivateMethodCheck.php

```diff
@@ -164,3 +164,89 @@ private function five(): void // Noncompliant
   {
   }
 }
+
+class MagicMethodCall
+{
+  public function __call($method, $arguments)
+  {
+    if (method_exists($this, $method)) {
+      return call_user_func_array([$this, $method], $arguments);
+    }
+    trigger_error('Call to undefined method '.__CLASS__.'::'.$method.'()', E_USER_ERROR);
+  }
+
+  // OK because it might be called via __call()
+  private function bar()
+  {
+  }
+}
+
+class MagicMethodCall2
+{
+  public function __call($method, $arguments)
+  {
+    return call_user_func($method, $arguments);
+  }
+
+  // OK because it might be called via __call()
+  private function bar()
+  {
+  }
+}
+
+class MagicMethodCall3Base
+{
+  public function __call($method, $arguments)
+  {
+    return call_user_func($method, $arguments);
+  }
+}
+
+class MagicMethodCall3Impl extends MagicMethodCall3Base
+{
+  // OK because it might be called via __call() in superclass
+  private function bar()
+  {
+  }
+}
+
+class MagicMethodCall4Base
+{
+  public function __call($method, $arguments)
+  {
+    return call_user_func($method, $arguments);
+  }
+}
+
+class MagicMethodCall4Abstract extends MagicMethodCall4Base {}
+
+class MagicMethodCall4Impl extends MagicMethodCall4Abstract
+{
+  // OK because it might be called via __call() in superclass
+  private function bar()
+  {
+  }
+}
+
+class MagicMethodCall5
```

Comment:
Maybe also add a test that would be FN for now? I.e., superclass has `__call` without `call_user_func`, but we don't raise an issue for a method in inheritor